components/CardDetails.tsx

```diff
@@ -0,0 +1,74 @@
+'use client';
+
+import React, { Fragment } from 'react';
+import Image from 'next/image';
+
+import { Dialog, Transition } from '@headlessui/react';
+
+interface CardDetailsProps {
+  isOpen: boolean;
+  closeModal: () => void;
+}
+
+const CardDetails = ({ isOpen, closeModal }: CardDetailsProps) => {
+  return (
+    <>
```

Comment:
Do we need the Fragment? `Transition` is the common parent of all the other nodes below

Reply:
I have refactor it, thank you